File: artifacts/lodepng/translate_new_ablation/src/lode_p_n_g_color_mode.rs
use crate::*;

#[derive(Clone)]
pub struct LodePNGColorMode {
    pub colortype: LodePNGColorType,
    pub bitdepth: u32,
    pub palette: Option<Vec<u8>>,
    pub palettesize: usize,
    pub key_defined: u32,
    pub key_r: u32,
    pub key_g: u32,
    pub key_b: u32,
}

File: artifacts/lodepng/decompose/src/lode_p_n_g_color_mode.c
typedef struct LodePNGColorMode
{
  LodePNGColorType colortype;
  unsigned bitdepth;
  unsigned char *palette;
  size_t palettesize;
  unsigned key_defined;
  unsigned key_r;
  unsigned key_g;
  unsigned key_b;
} LodePNGColorMode
/*** DEPENDENCIES:
typedef enum LodePNGColorType
{
  LCT_GREY = 0,
  LCT_RGB = 2,
  LCT_PALETTE = 3,
  LCT_GREY_ALPHA = 4,
  LCT_RGBA = 6,
  LCT_MAX_OCTET_VALUE = 255
} LodePNGColorType
----------------------------
***/


